Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleWindow.cpp

```diff
@@ -37,6 +38,52 @@ namespace ShaderManagementConsole
         Base::OnDocumentOpened(documentId);
         m_documentInspector->SetDocumentId(documentId);
     }
+
+    void ShaderManagementConsoleWindow::OnDocumentModified(const AZ::Uuid& documentId)
+    {
+        Base::OnDocumentModified(documentId);
+        m_documentInspector->SetDocumentId(documentId);
+    }
```

Comment:
This might cause a feedback loop and crash if you're editing from the table view or the inspector.

I would recommend using something like OnDocumentCreated but that might be sent before the document tab is added.

You could create another function to call from inside this callback

```
        documentTypeInfo.m_documentViewFactoryCallback = [this](const AZ::Crc32& toolId, const AZ::Uuid& documentId) {
            return m_window->AddDocumentTab(documentId, new ShaderManagementConsoleTableView(toolId, documentId, m_window.get()));
        };
```

You could also create another notification bus function OnDocumentViewCreated then add it here like this

```
    AZ::Uuid AtomToolsDocumentSystem::CreateDocumentFromType(const DocumentTypeInfo& documentType)
    {
        AZStd::unique_ptr<AtomToolsDocumentRequests> document(documentType.CreateDocument(m_toolId));
        if (!document)
        {
            DisplayErrorMessage(
                GetToolMainWindow(),
                QObject::tr("Document could not be created"),
                QObject::tr("Could not create document using type: %1").arg(documentType.m_documentTypeName.c_str()));
            return AZ::Uuid::CreateNull();
        }

        const AZ::Uuid documentId = document->GetId();
        m_documentMap.emplace(documentId, document.release());
        if (documentType.CreateDocumentView(m_toolId, documentId))
        {
            AtomToolsDocumentNotificationBus::Event(m_toolId, &AtomToolsDocumentNotificationBus::Events::OnDocumentViewCreated, documentId);
        }
        return documentId;
    }
```

Reply:
All I need is the document id when saving, GetCurrentDocumentId() works fine for me.